src/rgw/rgw_auth.cc

```diff
@@ -307,6 +330,23 @@ rgw::auth::Strategy::add_engine(const Control ctrl_flag,
   auth_stack.push_back(std::make_pair(std::cref(engine), ctrl_flag));
 }
 
+void
+rgw::auth::Strategy::add_engines(std::vector<std::string> auth_order,
```

Comment:
If we do target S3 only, then maybe having this machinery in `AWSAuthStrategy` is on option?

Reply:
While the implementation is only for the s3 engine right now, I thought there was potential for doing a similar strategy on swift engines as well. Otherwise I'm fine with moving all these down the the AWSAuth classes

Reply:
i think these 'auth names' should be specific to aws, so that swift could define its own. so this and parse_auth_order() are better as details of AWSAuthStrategy